packages/codec/lib/format/elementary.ts

```diff
@@ -64,7 +68,13 @@ export interface AddressValue {
   type: Types.AddressType;
   kind: "value";
   value: {
-    asAddress: string; //should have 0x and be checksum-cased
+    /**
+     * should have leading "0x" and be checksum-cased
```

Comment:
leading `0x` is probably "must" and checksum is probably a "should"

Reply:
They're both "must".  I'll just leave out words like "should".